Plus/ChicagoPlus.1.txt
NAME
  ChicagoPlus - convert Windows 95 theme to Xfce theme
SYNOPSIS
  ChicagoPlus [options] _filename_
OPTIONS
  <filename>            Theme file to convert
  -h, --help            show help message and exit
  -d, --debug           Print lots of debugging statements (default: 30)
  -v, --verbose         Be verbose (default: None)
  --noinstall           Do not place folders nor change theme (default: False)
  --info                Prints the json conversion of the theme file (default: False)

  -c COLORS, --colors COLORS  How many colors before skipping Inkscape fix/merge for SVGs. Set to 1 to speed up conversion. WARNING: This may result in transparent icons! (default: 32)

  -o OVERLAP, --overlap OVERLAP  Pixel overlap for SVG icons (default: 1)

  -s SQUARESIZE, --squaresize SQUARESIZE  Square size for SVG icons (default: 20)

  --cursorfolder CURSORFOLDER  Chicago95 cursor folder to convert to new theme (default: ~/.icons/Chicago95_Cursor_Black)

  --themefolder THEMEFOLDER  Chicago95 theme folder to convert to new theme (default: ~/.themes/Chicago95)

  --iconsfolder ICONSFOLDER  Chicago95 icons folder to convert to new theme (default: ~/.icons/Chicago95)

  --installdir INSTALLDIR  Folder to create new theme in, default is current working directory (default: . )

  --logfile LOGFILE     Filename for debug logging (default: chicago95_plus.log)

  SPECIFIC INSTALLS
  --nocursors           Do not generate/install/enable the cursors (default: True)
  --noicons             Do not generate/install/enable the icons (default: True)
  --nowallpaper         Do not generate/install/enable the wallpaper (default: True)
  --nosounds            Do not generate/install/enable the sounds (default: True)
  --nocolors            Do not generate/install/enable the colors (default: True)
  --nofonts             Do not generate/install/enable the fonts (default: True)
  --noscreensaver       Do not generate/install the screensaver (default: True)
DESCRIPTION
  Chicago95 Plus! is a python script that can parse Windows 95/98/ME/XP theme files and create new Chicago95 themes.
Chicago95 Plus! supports Icons, Cursors, Fonts, Wallpapers, Sounds and Theme colors! Use this against themes you can find on ThemeWorld or any site that have Microsoft Plus! Themes! This script can be called from any folder, execute it and provide it the path to a theme file (e.g. ChicagoPlus /home/bgates/Wicked/Wicked.theme) and your new theme will be created!
AUTHOR
  grassmunk, adapted as a manpage by bgstack15
COPYRIGHT
  GPL 3.0+
SEE ALSO
PlusGUI(1)


debian/changelog
chicago95 (0.0.1-1+devuan) obs; urgency=low

  * Initial release for Devuan. Features include:
    - Cursor themes
    - Xfce4-terminal color scheme
    - Startup sound
    - Fonts for terminal
    - Icon themes
    - Lightdm theme (lightdm-webkit2-greeter)
    - Plymouth splash screens
    - Gtk/metacity/xfce graphical themes
  * Special dependencies not available in stock Devuan:
    - lightdm-webkit2-greeter
    - plymouth
    - See README.Debian file for more info

 -- B Stack <bgstack15@gmail.com>  Tue, 10 Sep 2019 20:42:20 -0400


Plus/ChicagoPlus.py
#!/usr/bin/env python3

import sys
from pluslib import ChicagoPlus
import argparse
import logging
import os
from pathlib import Path

def main():
	

	desc = '''Chicago95 Plus! is a python script that can parse Windows 95/98/ME/XP theme files and create new Chicago95 themes. Chicago95 Plus! supports Icons, Cursors, Fonts, Wallpapers, Sounds and Theme colors! Use this against themes you can find on ThemeWorld or any site that have Microsoft Plus! Themes!\nThis script can be called from any folder, execute it and provide it the path to a theme file (e.g. %(prog)s /home/bgates/Wicked/Wicked.theme) and your new theme will be created!'''
	arg_parser = argparse.ArgumentParser(description=desc, 
						usage='%(prog)s [options] MS_Theme_File', 
						epilog="Part of the Chicago95 theme project",
						formatter_class=argparse.ArgumentDefaultsHelpFormatter)
	arg_parser.add_argument('-d', '--debug', help="Print lots of debugging statements", action="store_const", dest="loglevel", const=logging.DEBUG, default=logging.WARNING)
	arg_parser.add_argument('-v', '--verbose', help="Be verbose", action="store_const", dest="loglevel", const=logging.INFO)
	arg_parser.add_argument("--noinstall", help="Do not place folders nor change theme", action="store_true")
	arg_parser.add_argument("--info", help="Prints the json conversion of the theme file", action="store_true")
	arg_parser.add_argument('-c', '--colors', help='How many colors before skipping Inkscape fix/merge for SVGs. Set to 1 to speed up conversion. WARNING: This may result in transparent icons!', default=32, type=int)
	arg_parser.add_argument('-o', '--overlap', help='Pixel overlap for SVG icons', default=1, type=int)	
	arg_parser.add_argument('-s', '--squaresize', help='Square size for SVG icons', default=20, type=int)
	arg_parser.add_argument('--cursorfolder', help="Chicago95 cursor folder to convert to new theme", default=str(Path.home())+"/.icons/Chicago95_Cursor_Black")
	arg_parser.add_argument('--themefolder', help="Chicago95 theme folder to convert to new theme", default=str(Path.home())+"/.themes/Chicago95")
	arg_parser.add_argument('--iconsfolder', help="Chicago95 icons folder to convert to new theme", default=str(Path.home())+"/.icons/Chicago95")
	arg_parser.add_argument("--installdir", help="Folder to create new theme in, default is current working directory", default=os.getcwd())
	arg_parser.add_argument("--logfile", help="Filename for debug logging", default="chicago95_plus.log")
	arg_parser.add_argument("theme_file", help="Microsoft Windows 95/98/ME .theme file")
	specific_args = arg_parser.add_argument_group("Specific Installs")
	specific_args.add_argument("--nocursors", help="Do not generate/install/enable the cursors", action="store_false")
	specific_args.add_argument("--noicons", help="Do not generate/install/enable the icons", action="store_false")
	specific_args.add_argument("--nowallpaper", help="Do not generate/install/enable the wallpaper", action="store_false")
	specific_args.add_argument("--nosounds", help="Do not generate/install/enable the sounds", action="store_false")
	specific_args.add_argument("--nocolors", help="Do not generate/install/enable the colors", action="store_false")
	specific_args.add_argument("--nofonts", help="Do not generate/install/enable the fonts", action="store_false")
	specific_args.add_argument("--noscreensaver", help="Do not generate/install the screensaver", action="store_false")

	args = arg_parser.parse_args()	

	plus = ChicagoPlus(themefile=args.theme_file, 
			   loglevel=args.loglevel, 
			   colors=args.colors, 
			   overlap=args.overlap, 
			   squaresize=args.squaresize, 
			   installdir=args.installdir,
			   chicago95_cursor_path=args.cursorfolder,
			   chicago95_theme_path=args.themefolder,
			   chicago95_icons_path=args.iconsfolder,
			   logfile=args.logfile)

	print(plus.logo())

	if args.info:
		plus.parse_theme()
		plus.print_theme_config()
		sys.exit(0)

	if args.noinstall:
		plus.check_software(args.nocursors, args.noicons, args.colors)
		plus.parse_theme()
		plus.generate_theme(cursors=args.nocursors, icons=args.noicons, wallpaper=args.nowallpaper, sounds=args.nosounds, colors=args.nocolors, fonts=args.nofonts, screensaver=args.noscreensaver)
	else:
		plus.go(cursors=args.nocursors, icons=args.noicons, wallpaper=args.nowallpaper, sounds=args.nosounds, colors=args.nocolors, fonts=args.nofonts, screensaver=args.noscreensaver)	

main()
